### PR TITLE
Add streaming result sets support

### PR DESCRIPTION
**Pull request by bserdar**
Stream result sets instead of storing them in lists

### CODE REVIEW

File: mongo/src/main/java/com/redhat/lightblue/mongo/crud/IterateAndUpdate.java [hunk 1]
```diff
@@ -172,7 +175,6 @@ public void update(CRUDOperationContext ctx,
             cursor.setReadPreference(ReadPreference.primary());
             measure.end("collection.find");
             LOGGER.debug("Found {} documents", cursor.count());
-            ctx.getFactory().getInterceptors().callInterceptors(InterceptPoint.POST_CRUD_UPDATE_RESULTSET, ctx);
```
Comment (bvulaj) [inline mongo/src/main/java/com/redhat/lightblue/mongo/crud/IterateAndUpdate.java]: Keeping pre_crud but ditching post_crud?

**Comment by bserdar** [comment]
Yes, post_* screws everything up, and it is not used anyway. With post_*, I
can't stream, because post_* interceptors need access to the whole
resultset.

**Comment by paterczm** [comment]
Isn't post update hook used for integration? @alechenninger, you'll know.

**Comment by bserdar** [comment]
This doesn't change the hook processing. All hooks are post_ hooks.

**Comment by paterczm** [comment]
Hooks != interceptors. So never mind.

**Pull request merged**